### PR TITLE
Port vicenc edward material editor - part 1

### PR DESCRIPTION
**Pull request by VBoschMatas**
Part 1 of integration. Part 2 will be done after the Resources are completed.

- Added UI enhancements to the material component. 
- Added name var "std::string" to materials.

### CODE REVIEW

**Comment by Bernatmago** [comment]
Task linked: [CU-1ykabuf Port Vicenç & Edward material editor](https://app.clickup.com/t/1ykabuf)

**Review by baransrc — CHANGES_REQUESTED** [review]
I requested a small change, aside from that everything looks good. Also it's better if you can amend your last commit and delete the tmp_scene.scene, and then force-push. Good job my dude!

File: Source/src/components/ComponentMaterial.cpp [hunk 1]
```diff
@@ -34,37 +34,87 @@ void Hachiko::ComponentMaterial::Load(JsonFormatterValue j_component)
 
 void Hachiko::ComponentMaterial::DrawGui()
 {
+    static ImGuiTreeNodeFlags texture_flags = ImGuiTreeNodeFlags_SpanAvailWidth | ImGuiTreeNodeFlags_DefaultOpen;
```
[the remark this answers is not in the document]
Comment (baransrc) [reply Source/src/components/ComponentMaterial.cpp:37]: It's better to have this const as the flags are not updated below,

**Review by baransrc — APPROVED** [review]
Everything looks good, thanks for your time and effort!!

**Pull request merged**